doc/decisions/1_in_discussion/change_tracking.md

```diff
@@ -124,15 +139,13 @@ Downsides of this approach:
 Use the `backendData->keys` for change tracking
 
 We already store which keys have been returned by `kdbGet` for each backend within KDB.
-Currently, however, this is not a deep duplication, as we are returning the keys from `backendData->keys` directly.
+Currently, however, this is not a deep copy, as we are returning the keys from `backendData->keys` directly.
 This means we can not detect changes to the values or metadata of keys right now.
 We can, however, rely on this for detecting removed and added keys in its current form.
 
-If we don't want to deep-dup it, we'd need to do something different to detect which keys have been modified.
-One possibility would be to add metadata within the `key` functions (e.g. `meta:/elektra/original`), but that would violate the `libelektra-core` must be minimal constraint.
+As we now implement copy-on-write for all keys and keysets within Elektra, we can deep copy what we return in `kdbGet`.
 
-There is already [another decision](../1_in_discussion/copy_on_write.md) which discusses adding general copy-on-write semantics to Elektra.
-We could use this together with `backendData->keys` deep-duped to do memory efficient change tracking.
+Another problem with this approach is that `backendData->keys` get cleared every time `kdbGet` is called.
```

Comment:
Again, please avoid internals like `backendData->keys` (or introduce them, as not everyone knows the details of new-backend yet).

Reply:
```suggestion
Another problem with this approach is that the internally stored keys are recreated as new instances every time `kdbGet` is called.
```

Not quite true I think, but good enough here.